### PR TITLE
Add docs about Kafka Connect and Apache Camel

### PR DESCRIPTION
**Pull request by eolivelli**

### CODE REVIEW

File: configuration-resources/connectors/camel.md [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+    - name: "Apache Camel GitHub component - dependency"
+      url: "https://repo1.maven.org/maven2/org/eclipse/mylyn/github/org.eclipse.egit.github.core/2.1.5/org.eclipse.egit.github.core-2.1.5.jar"
+      sha512sum: "4fab14efed656b6705ad74ab9c05a346fe4fdc737b1b5130842aee7ebba98b895b7cb8c35d15e3da150116f445ae3b5c88daaaa5319c39144cb365247b9d0630"
+      type: "java-library"
+```
+
+You have to bundle all the jars needed by the connector, in this example the Apache Camel GitHub component requires the org.eclipse.egit.github.core jar file.
+
+The jar files are downloaded by the LangStream CLI when you are deploying the application and then copied to the java/lib directory.
+You are not required to use this mechanism, you can copy manuallly the jar files.
```
Comment (cbornet) [inline configuration-resources/connectors/camel.md:27]:
```suggestion
You are not required to use this mechanism, you can copy manually the jar files.
```

File: configuration-resources/connectors/kafka-connect.md [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+configuration:
+  dependencies:
+    - name: "Kafka Connect Sink for Apache Cassandra from DataStax"
+      url: "https://github.com/datastax/kafka-sink/releases/download/1.5.0/kafka-connect-cassandra-sink-1.5.0.jar"
+      sha512sum: "242bf60363d36bd426232451cac836a24ae8d510857372a128f601503ad77aa9eabf14c4f484ca0830b6a68d9e8664e3820739ad8dd3deee2c58e49a94a20a3c"
+      type: "java-library"
+```
+
+The jar file is downloaded by the LangStream CLI when you are deploying the application and it is copied to the java/lib directory.
+You are not required to use this mechanism, you can copy manuallly the jar file.
```
Comment (cbornet) [inline configuration-resources/connectors/kafka-connect.md:21]:
```suggestion
You are not required to use this mechanism, you can copy manually the jar file.
```

File: configuration-resources/connectors/kafka-connect.md [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+          creation-mode: create-if-not-exists
+        - name: "offset-topic"
+          creation-mode: create-if-not-exists
+          partitions: 1
+          options:
+              replication-factor: 1
+          config:
+              cleanup.policy: compact
+    pipeline:
+        - name: "source1"
```
Comment (nicoloboschi) [inline configuration-resources/connectors/kafka-connect.md:84]: indentation

**Review by nicoloboschi — CHANGES_REQUESTED** [review]

File: configuration-resources/connectors/camel.md [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+But if you use the dependency mechanism, the LangStream CLI will check the sha512sum of the files to make sure that they are not corrupted.
+
+It is suggested to add a .gitignore file into your application in order to not commit the jar file into your git repository.
+
+
+### Apache Camel Sources
+
+Once you have your connector deployed into your application, you can use it in your pipeline.
+
+This is an example about how to configure a Source that reads events from GitHub
```
Comment (mendonk) [inline configuration-resources/connectors/camel.md:37]:
```suggestion
This is an example of configuring a Source connector that reads events from GitHub:
```

File: configuration-resources/connectors/camel.md [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+  - name: "keep only the contents of the comment"
+    type: "compute"
+    output: "output-topic"
+    configuration:
+        fields:
+           - name: "value"
+             expression: "value.body"
+```
+
+In the "configuration" section you must provide the component-uri and the component-options.
```
Comment (mendonk) [inline configuration-resources/connectors/camel.md:62]:
```suggestion
You must provide the component-uri and the component-options in the "configuration" section.
```

File: configuration-resources/connectors/camel.md [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+
+All the component-options are passed to the component as additional parameters in the query string, appended to the component-uri.
+This mechanism helps you in defining each property in a separate secret, so that you can easily rotate the secrets without changing the pipeline configuration.
+Also The camel-source agent will automatically URL encode the values of the parameters passed to the querystring.
```
Comment (mendonk) [inline configuration-resources/connectors/camel.md:66]:
```suggestion
Additionally, the camel-source agent will automatically URL-encode the values of the parameters passed to the query string.
```

File: configuration-resources/connectors/kafka-connect.md [hunk 7]
```diff
@@ -0,0 +1,92 @@
...
+  dependencies:
+    - name: "Kafka Connect Sink for Apache Cassandra from DataStax"
+      url: "https://github.com/datastax/kafka-sink/releases/download/1.5.0/kafka-connect-cassandra-sink-1.5.0.jar"
+      sha512sum: "242bf60363d36bd426232451cac836a24ae8d510857372a128f601503ad77aa9eabf14c4f484ca0830b6a68d9e8664e3820739ad8dd3deee2c58e49a94a20a3c"
+      type: "java-library"
+```
+
+The jar file is downloaded by the LangStream CLI when you are deploying the application and it is copied to the java/lib directory.
+You are not required to use this mechanism, you can copy manually the jar file.
+But if you use the dependency mechanism, the LangStream CLI will check the sha512sum of the file to make sure that the file is not corrupted.
```
Comment (mendonk) [inline configuration-resources/connectors/kafka-connect.md:22]:
```suggestion
```

File: configuration-resources/connectors/kafka-connect.md [hunk 8]
```diff
@@ -0,0 +1,92 @@
...
+
+It is suggested to add a .gitignore file into your application in order to not commit the jar file into your git repository.
```
Comment (mendonk) [inline configuration-resources/connectors/kafka-connect.md:24]:
```suggestion
We recommend adding a .gitignore file into your application so you don't commit the jar file into your git repository.
```

File: configuration-resources/connectors/kafka-connect.md [hunk 9]
```diff
@@ -0,0 +1,92 @@
...
+
+
+### Kafka Connect Sinks
+
+Once you have your connector deployed into your application, you can use it in your pipeline.
+
+This is an example about how to configure a Sink that writes to Apache Cassandra.
```
Comment (mendonk) [inline configuration-resources/connectors/kafka-connect.md:31]:
```suggestion
This is an example of configuring a Sink connector that writes to Apache Cassandra:
```

File: configuration-resources/connectors/kafka-connect.md [hunk 10]
```diff
@@ -0,0 +1,92 @@
...
+      key.converter: org.apache.kafka.connect.storage.StringConverter
+      value.converter: org.apache.kafka.connect.storage.StringConverter
+      # Connector specific configuration
+      cloud.secureConnectBundle: "${ secrets.cassandra.secure-connect-bundle }"
+      auth.username: "${ secrets.cassandra.username }"
+      auth.password: "${ secrets.cassandra.password }"
+      topic.input-topic.vsearch.products.mapping: "id=value.id,description=value.description,name=value.name"
+```
+
+In the "configuration" section you can provide the configuration for the connector.
```
Comment (mendonk) [inline configuration-resources/connectors/kafka-connect.md:56]:
```suggestion
Provide the configuration for the connector in the "configuration" section of the yaml file.
```

File: configuration-resources/connectors/kafka-connect.md [hunk 11]
```diff
@@ -0,0 +1,92 @@
...
+you must provide connector.class, name, key.converter and value.converter, like for any other Kafka Connect connector.
```
Comment (mendonk) [inline configuration-resources/connectors/kafka-connect.md:57]:
```suggestion
You must provide `connector.class`, `name`, `key.converter` and `value.converter`, as you would for any other Kafka Connect connector.
```

File: configuration-resources/connectors/kafka-connect.md [hunk 12]
```diff
@@ -0,0 +1,92 @@
...
+Check you the reference documentation of the connector you are using for more details about the configuration properties.
```
Comment (mendonk) [inline configuration-resources/connectors/kafka-connect.md:58]:
```suggestion
Check the reference documentation of the connector you are using for more details about its configuration properties.
```

File: configuration-resources/connectors/kafka-connect.md [hunk 13]
```diff
@@ -0,0 +1,92 @@
...
+
+### Kafka Connect Sources
+
+Kafka Connect sources works the same way as sinks but you have to provide an additional system topic that Kafka Connect uses to store
```
Comment (mendonk) [inline configuration-resources/connectors/kafka-connect.md:62]:
```suggestion
Kafka Connect source connectors work the same way as sinks, but you have to provide an additional system topic that Kafka Connect uses to store the state of the source connector.
```

File: configuration-resources/connectors/kafka-connect.md [hunk 14]
```diff
@@ -0,0 +1,92 @@
...
+the state of the source.
```
Comment (mendonk) [inline configuration-resources/connectors/kafka-connect.md:63]:
```suggestion
```

File: configuration-resources/connectors/kafka-connect.md [hunk 15]
```diff
@@ -0,0 +1,92 @@
...
+In this example it is named "offset-topic".
```
Comment (mendonk) [inline configuration-resources/connectors/kafka-connect.md:64]:
```suggestion
In this example the system topic is named "offset-topic".
```

File: configuration-resources/connectors/kafka-connect.md [hunk 16]
```diff
@@ -0,0 +1,92 @@
...
+
+In order to make Kafka Connect runtime happy you have to configure the cleanup.policy to "compact" for the offset-topic.
```
Comment (mendonk) [inline configuration-resources/connectors/kafka-connect.md:66]:
```suggestion
To make the Kafka Connect runtime happy you have to set the `cleanup.policy` config value to "compact" for the offset-topic.
```

**Review by mendonk — APPROVED** [review]

**Pull request merged**